docs/app/Components/ComponentDoc/ComponentDoc.js

```diff
@@ -0,0 +1,31 @@
+import React, {Component, PropTypes} from 'react';
+import {Segment} from 'stardust';
+
+import ComponentDescription from './ComponentDescription';
+import ComponentExamples from './ComponentExamples';
+import ComponentProps from './ComponentProps';
+import getComponentDocInfo from 'docs/app/utils/GetComponentDocInfo.js';
+
+export default class ComponentDoc extends Component {
+  static propTypes = {
+    name: PropTypes.string,
+  };
+
+  render() {
+    let doc = getComponentDocInfo(this.props.name);
+
+    return (
+      <Segment id={doc.name}>
+        <ComponentDescription
+          path={doc.path}
+          name={doc.name}
+          parent={doc.parent}
+          type={doc.type}
+          description={doc.docBlock.description}
+        />
+        <ComponentProps props={doc.props} />
+        <ComponentExamples name={doc.name} />
+      </Segment>
+    );
+  }
+}
```

Comment:
This component is what generates a doc section for a given component.  Every component in stardust is looped and passed through this component to generate its doc.